src/core/client_config/uri_parser.c

```diff
@@ -99,7 +99,7 @@ grpc_uri *grpc_uri_parse(const char *uri_text, int suppress_errors) {
   if (uri_text[scheme_end + 1] == '/' && uri_text[scheme_end + 2] == '/') {
     authority_begin = scheme_end + 3;
     for (i = authority_begin; uri_text[i] != 0; i++) {
-      if (uri_text[i] == '/') {
+      if (uri_text[i] == '/' && authority_end == -1) {
```

Comment:
Wouldn't it be better to put the `&& authority_end == -1` clause in the for loop, to make its termination condition `uri_text[i] != 0 && authority_end == -1`?